test/apps/iodemo/io_demo.cc

```diff
@@ -1849,20 +1863,26 @@ class DemoClient : public P2pDemoCommon {
             op_info[op].num_iters++;
 
             if (is_control_iter(total_iter) && (total_iter > total_prev_iter)) {
-                /* Print performance every 1 second */
-                double curr_time = get_time();
-                if (curr_time >= (prev_time + opts().print_interval)) {
-                    wait_for_responses(0);
-                    if (_status != OK) {
-                        break;
-                    }
+                if ((op_info[IO_WRITE].num_iters +
+                             op_info[IO_READ].num_iters) >=
+                    (_num_active_servers_to_use * opts().operations.size())) {
+                    /* Print performance every 1 second */
+                    double curr_time = get_time();
+                    if (curr_time >= (prev_time + opts().print_interval)) {
+                        wait_for_responses(0);
+                        if (_status != OK) {
+                            break;
+                        }
 
-                    report_performance(total_iter - total_prev_iter,
-                                       curr_time - prev_time, op_info);
-                    total_prev_iter = total_iter;
-                    prev_time       = curr_time;
+                        report_performance(total_iter - total_prev_iter,
+                                           curr_time - prev_time, op_info);
+                        total_prev_iter = total_iter;
+                        prev_time       = curr_time;
 
-                    check_time_limit(curr_time);
+                        check_time_limit(curr_time);
+                    }
+                } else {
+                    progress();
```

Comment:
would not `wait_for_responces` below be enough?

Reply:
`wait_for_responces(0)` after the loop which does all operations during the testing.
so, we should call `progress()` from time to time.

Reply:
we have another one above.

Reply:
the `wait_for_responses(max_outstanding)` will do progress and wait for responses only if `number of sent - number of completed` > `window size` which is false in our case.

Reply:
If we have connected (or connecting to) servers, we are posting OPs above, so `wait_for_responses` should call progress, when window is full. Case when all connections are down is handled above, could you please describe which case is missed?

Reply:
so, it's better or we must do progress?
if "better" then why?
In my understanding, if window is not full we can post more and then progress in `wait_for_responses`